config/cmake/H5pubconf.h.in

```diff
@@ -190,9 +190,6 @@
 /* Define to 1 if you have the `curl' library (-lcurl). */
 #cmakedefine H5_HAVE_LIBCURL @H5_HAVE_LIBCURL@
 
-/* Define to 1 if you have the `dl' library (-ldl). */
```

Comment:
We still maintain two build systems: CMake and autotools/autoconf.
This file needs to remain in sync with the file generated by autoconf until such time that we no longer support autoconf.

Reply:
Nothing cares if this define is present or not, so it isn't doing anything in the header.

Reply:
We keep this file in sync with the one produced by Autoconf so it's easier to eyeball them to make sure we aren't missing something in the other build system.

Reply:
A lot of these header files are useless, though, and have already been removed from configure.ac.